Sources/Brave/Frontend/Shields/SubmitReportView.swift

```diff
@@ -0,0 +1,95 @@
+// Copyright 2023 The Brave Authors. All rights reserved.
+// This Source Code Form is subject to the terms of the Mozilla Public
+// License, v. 2.0. If a copy of the MPL was not distributed with this
+// file, You can obtain one at https://mozilla.org/MPL/2.0/.
+
+import SwiftUI
+import Strings
+import BraveUI
+import DesignSystem
+
+struct SubmitReportView: View {
+  let url: URL
+  let dismiss: () -> Void
+  let submit: (URL, String, String) -> Void
+  
+  @State private var additionalDetails = ""
+  @State private var contactDetails = ""
+  
+  private var scrollContent: some View {
+    ScrollView {
+      VStack(alignment: .leading, spacing: 16) {
+        Text(Strings.Shields.reportBrokenSiteBody1)
+        Text(url.absoluteString)
+          .foregroundStyle(Color(braveSystemName: .textInteractive))
+        Text(Strings.Shields.reportBrokenSiteBody2)
+          .font(.footnote)
+        BraveTextEditor(
+          text: $additionalDetails,
+          prompt: Strings.Shields.reportBrokenAdditionalDetails
+        ).frame(height: 80)
+        
+        VStack(alignment: .leading, spacing: 4) {
+          Text(Strings.Shields.reportBrokenContactMe).font(.caption)
+          TextField(
+            Strings.Shields.reportBrokenContactMe,
+            text: $contactDetails, prompt: Text(Strings.Shields.reportBrokenContactMeSuggestions)
+          )
+          .textFieldStyle(BraveTextFieldStyle())
+          .autocorrectionDisabled()
+          .textInputAutocapitalization(.never)
+        }
+      }
+      .padding()
+      .foregroundStyle(Color(braveSystemName: .textSecondary))
+    }
+    .foregroundStyle(Color(braveSystemName: .textSecondary))
+    .navigationTitle(Strings.Shields.reportABrokenSite)
+    .toolbar {
+      ToolbarItem(placement: .cancellationAction) {
+        Button(Strings.cancelButtonTitle, action: dismiss)
+      }
+      ToolbarItem(placement: .confirmationAction) {
+        Button(Strings.Shields.reportBrokenSubmitButtonTitle, action: {
+          didTapSubmit()
+        })
+      }
+    }
+  }
+  
+  var body: some View {
+    if #available(iOS 17.0, *) {
+      NavigationStack {
+        #if swift(>=5.9)
+        scrollContent
+          .toolbarTitleDisplayMode(.inline)
+        #else
+        scrollContent
+        #endif
+      }
+    } else if #available(iOS 16.0, *) {
+      NavigationStack {
+        scrollContent
+      }
+    } else {
+      NavigationView {
+        scrollContent
+      }
+    }
```

Comment:
No need to do any of this

1. Just use `NavigationView` + `.navigationViewStyle(.stack)` on all versions
2. Use `navigationTitleDisplayMode` on all versions

Reply:
Oh awesome, thanks for the tip